src/metabase/util.cljc

```diff
@@ -944,3 +947,27 @@
    (fn [acc x] (if (pred x) (reduced x) acc))
    nil
    coll))
+
+#?(:clj
+   (let [sym->enum (fn ^Enum [sym]
+                    (Reflector/invokeStaticMethod ^Class (resolve (symbol (namespace sym)))
+                                                  "valueOf"
+                                                  (to-array [(name sym)])))
+         ordinal (fn [^Enum e] (.ordinal e))]
+     (defmacro case-enum
+       "Like `case`, but explicitly dispatch on Java enum ordinals.
+
+       Passing the same enum type as the ones you're checking in is on you, this is not checked."
+       [value & clauses]
+       (let [types (map (comp type sym->enum first) (partition 2 clauses))]
```

Comment:
nit: You could calculate the type a bit more directly with just `(comp resolve symbol namespace first)`